Dart/src/com/jetbrains/lang/dart/flutter/FlutterUtil.java

```diff
@@ -8,27 +10,47 @@
 public class FlutterUtil {
 
   private static final String FLUTTER_MODULE_TYPE_ID = "FLUTTER_MODULE_TYPE";
+  private static PluginId FLUTTER_PLUGIN_ID = PluginId.getId("io.flutter");
+
+  /**
+   * Get the Flutter root relative to the given Dart SDK.
+   *
+   * @param dartSdk the Dart SDK
+   * @return the relative Flutter root, or null if this SDK is not relative to one
+   */
+  @Nullable
+  public static String getFlutterRoot(@NotNull final String dartSdkPath) {
+    final String suffix = "/bin/cache/dart-sdk";
+    return dartSdkPath.endsWith(suffix) ? dartSdkPath.substring(0, dartSdkPath.length() - suffix.length()) : null;
+  }
+
+  /**
+   * Test if the given project defines a Flutter module.
+   *
+   * @param module the project to test
+   * @return true if the given project has a Flutter module, false otherwise.
+   */
+  public static boolean hasFlutterModule(@NotNull project) {
+    return Arrays.stream(ModuleManager.getInstance(project).getModules()).anyMatch(FlutterUtil::isFlutterModule);
```

Comment:
`ModuleUtil.hasModulesOfType` does it faster. You'll probably need to use `ModuleTypeManager.findByID` as well.